src/shared/ui/Button/index.tsx

```diff
@@ -21,21 +22,47 @@ export const buttonVariants = cva(['font-sans font-bold'], {
   },
 });
 
+type IconPositionType = 'start' | 'end';
+
+const iconVariants = cva('absolute top-1/2 -translate-x-1/2 -translate-y-1/2', {
+  variants: {
+    iconPosition: {
+      start: 'left-7',
+      end: 'right-5',
+    },
+  },
+  defaultVariants: {
+    iconPosition: 'start',
+  },
+});
+
 export interface ButtonProps
   extends ButtonHTMLAttributes<HTMLButtonElement>,
-    VariantProps<typeof buttonVariants> {
+    VariantProps<typeof buttonVariants>,
+    VariantProps<typeof iconVariants> {
   title: string;
+  iconPosition?: IconPositionType;
+  icon?: ReactNode;
   onClick?: () => void;
   className?: string;
+  isSubmit?: boolean;
 }
 
-export const Button: FC<ButtonProps> = ({ variant, size, ...props }) => {
+export const Button: FC<ButtonProps> = ({
+  variant,
+  size,
+  icon,
+  className,
+  isSubmit = false,
+  ...props
+}) => {
   return (
     <button
-      type='button'
+      type={isSubmit ? 'submit' : 'button'}
       onClick={props.onClick}
-      className={cn(buttonVariants({ variant, size }), props.className)}
+      className={cn(buttonVariants({ variant, size, className }))}
     >
+      <div>{icon}</div>
```

Comment:
```suggestion
      {icon}
```
This is potential empty div